crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -1668,12 +1669,19 @@ impl ValueToString for NullFileSystem {
     }
 }
 
-pub async fn to_sys_path(path: FileSystemPathVc) -> Result<Option<PathBuf>> {
-    if let Some(fs) = DiskFileSystemVc::resolve_from(path.fs()).await? {
-        let sys_path = fs.await?.to_sys_path(path).await?;
-        return Ok(Some(sys_path));
+pub async fn to_sys_path(mut path: FileSystemPathVc) -> Result<Option<PathBuf>> {
+    loop {
```

Comment:
It looks like this can only loop 1 time. Did you mean to continue if we unwrapped an `AttachedFileSystemVc`?

Reply:
yes, fortunately clippy also found this